unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/BoxShape/ECSBoxShapeComponentHandler.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using DCL;
+using DCL.Controllers;
+using DCL.ECSRuntime;
+using DCL.Helpers;
+using DCL.Models;
+using UnityEngine;
+
+public class ECSBoxShapeComponentHandler : IECSComponentHandler<ECSBoxShape>
+{
+    private AssetPromise_PrimitiveMesh primitiveMeshPromisePrimitive;
+    internal MeshesInfo meshesInfo;
+
+    private bool isDisposed = false;
+    
+    public void OnComponentCreated(IParcelScene scene, IDCLEntity entity) { }
+    
+    public void OnComponentRemoved(IParcelScene scene, IDCLEntity entity)
+    {
+        Dispose(scene);
+    }
+
+    public void Dispose(IParcelScene scene)
+    {
+        if (isDisposed)
+            return;
+        
+        isDisposed = true;
+        
+        if (primitiveMeshPromisePrimitive != null)
+            AssetPromiseKeeper_PrimitiveMesh.i.Forget(primitiveMeshPromisePrimitive);
+        
+        Utils.CleanMaterials(meshesInfo.meshRootGameObject.GetComponent<Renderer>());
+        meshesInfo.CleanReferences();
+        ECSComponentsUtils.RemoveRendereableFromDataStore(scene.sceneData.id,meshesInfo.rendereable);
```

Comment:
could we move this call before `meshsesInfo.CleanReferences`?

Reply:
yep, we can